mock.go

```diff
@@ -569,25 +570,36 @@ type Call struct {
 	complete chan struct{}
 }
 
+// Call represents an apiCall that has been trapped.
+type Call struct {
+	Time     time.Time
+	Duration time.Duration
+	Tags     []string
+
+	apiCall *apiCall
+	trap    *Trap
+}
+
 func (c *Call) Release() {
-	c.releases.Done()
-	<-c.complete
+	c.apiCall.releases.Done()
+	<-c.apiCall.complete
```

Comment:
Wish we could use `t.Context()` here but I did some experiments and unless the use-case is very specific, it won't help at all (`go test -timeout=5s` for instance does not cancel the context before panicing).

Reply:
Agreed, alas.